shell/platform.ts

```diff
@@ -0,0 +1,12 @@
+import { EOL } from '../deps.ts';
+
+/** Any type of OS supported by Deno and potentially returned by {@link Deno.build.os}. */
+export type OsType = typeof Deno.build.os;
```

Comment:
Nit - I don't think it makes sense to export this- users should be using the built in `Deno.build.os` here instead of `OsType`.

Reply:
(case in point- all of the rest of the code in Rex)

Reply:
`Deno.build.os` just defines its type as `"darwin" | "linux" | "windows"` though, rather than a formal type. I dislike that for a couple reasons:

1. It's long and annoying to type
2. Using it other places presents the possible future where the type changes and you have many places that have to change as well

> (case in point- all of the rest of the code in Rex)
There's currently nowhere in the library where we define a function parameter or variable that will get assigned a `Deno.build.os` value, so there's nowhere to use this type alias. I do have other code that uses a version of this that I want to change to use Rex, it's just not in this PR.

Here's a snippet of a case where that happens:
```typescript
function createPackage(files: string[], targetOs: OsType)
```

We don't *need* to introduce this type though, and if you insist I'll remove it.

Reply:
I guess I don't feel strongly other than my general dislike of renaming/shortcutting/remapping of ecosystem things (even if it is longer to type), because they are one more level of abstraction that isn't adding any value and just requires new developers to learn one more thing.

Feel free to leave it in if it's useful.